components/hab/src/cli/hab/util.rs

```diff
@@ -232,6 +235,48 @@ impl fmt::Display for DurationProxy {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result { write!(f, "{}", u64::from(*self)) }
 }
 
+/// A wrapper around `SocketAddr`
+#[derive(Copy, Clone, Debug, PartialEq, Serialize, Deserialize)]
+#[serde(try_from = "String")]
+pub struct SocketAddrProxy (SocketAddr);
+
+impl TryFrom<String> for SocketAddrProxy {
+    type Error = Error;
+    fn try_from(value: String) -> Result<Self, Self::Error> {
+        let (domain, addr) =
+            habitat_common::util::resolve_socket_addr_with_default_port(value, GossipListenAddr::DEFAULT_PORT)?;
+        if !domain.is_empty() {
```

Comment:
I was looking at `resolve_socket_addr_with_default_port` and couldn't figure out under what condition `domain` would be empty.

Reply:
Agree. Have removed the unnecessary if check in commit 8930f52d98b666253fb494c288a5c2cdb3bf686c.